### PR TITLE
fix(diffviewer): handle undefined values to prevent TypeError (ENG-2188)

### PR DESCRIPTION
**Pull request by K-Mistele**
## What problem(s) was I solving?

Fixed a critical runtime error in the DiffViewer component that occurred when processing malformed tool input JSON with undefined or null values. The error "TypeError: undefined is not an object (evaluating 't.split')" was causing the CodeLayer UI to crash when backend services sent incomplete tool input data.

Related issues:
- [ENG-2188](https://linear.app/humanlayer/issue/ENG-2188/typeerror-undefined-is-not-an-object-evaluating-tsplit): TypeError: undefined is not an object (evaluating 't.split')
- [Sentry Issue](https://humanlayer-00.sentry.io/issues/6905794763/?referrer=Linear): Production error tracking

### Error Prevention
- DiffViewer component no longer crashes when receiving undefined/null content
- Malformed tool inputs now display empty diffs instead of causing TypeErrors
- UI remains stable even when backend sends incomplete JSON data

### Affected Components
- **Edit tool**: Handles missing `old_string`/`new_string` fields gracefully
- **Write tool**: Handles missing `content` field gracefully
- **MultiEdit tool**: Handles missing fields in edit arrays gracefully
- **DiffViewer**: Shows empty or partial diffs when content is undefined

### Core Fix (`DiffViewer.tsx`)
Added defensive null checks in the `computeLineDiff` function at lines 35-36:
```typescript
const oldLines = (oldStr || '').split('\n')
const newLines = (newStr || '').split('\n')
```

### Component Protection
Added fallback values in all components that pass props to DiffViewer:
- **EditToolCallContent.tsx**: Added `|| ''` fallbacks for `old_string` and `new_string`
- **WriteToolCallContent.tsx**: Added `|| ''` fallback for `content`
- **MultiEditToolCallContent.tsx**: Added `|| ''` fallbacks in map operations for edit arrays

### Testing Infrastructure
- Created comprehensive unit tests (`DiffViewer.test.tsx`) covering undefined/null edge cases
- Added Storybook stories for visual testing of edge cases
- Created integration tests to verify malformed tool inputs are handled properly
- Added test script (`test-malformed-inputs.sh`) for database injection testing

## How to verify it

- [x] I have ensured `make check test` passes (Note: 4 unrelated Python async tests are failing)

### Automated Testing
- [x] Unit tests pass: `cd humanlayer-wui && bun test DiffViewer.test`
- [x] Integration tests pass: `cd humanlayer-wui && bun test integration.test`
- [x] Type checking passes: `make -C humanlayer-wui check`
- [x] Build succeeds: `cd humanlayer-wui && bun run build`

### Manual Testing
1. Test malformed inputs:
   - Run `bash humanlayer-wui/scripts/test-malformed-inputs.sh` with an active session
   - Verify no TypeErrors appear in CodeLayer UI console
   - Check that empty diffs display correctly

2. Storybook testing:
   - View the new edge case stories: `UndefinedContent`, `PartiallyUndefinedOld`, `PartiallyUndefinedNew`
   - Verify they render without errors
   - Confirm partial content displays appropriately

3. Normal operation:
   - Verify existing Edit, Write, and MultiEdit tools continue working normally
   - Confirm no regression in diff visualization for valid inputs

## Description for the changelog

Fixed critical TypeError in DiffViewer component when receiving undefined or null values from malformed tool input JSON. Added defensive programming and comprehensive tests to prevent UI crashes from incomplete backend data.

----

> [!IMPORTANT]
> Fixes `DiffViewer` crash on undefined/null values by adding null checks and comprehensive tests.
> 
>   - **Behavior**:
>     - `DiffViewer` component now handles undefined/null values without crashing, displaying empty diffs instead.
>     - `EditToolCallContent`, `WriteToolCallContent`, and `MultiEditToolCallContent` components handle missing fields gracefully.
>   - **Core Fix**:
>     - Added null checks in `computeLineDiff` function in `DiffViewer.tsx`.
>   - **Component Protection**:
>     - Added `|| ''` fallbacks in `EditToolCallContent.tsx`, `WriteToolCallContent.tsx`, and `MultiEditToolCallContent.tsx`.
>   - **Testing**:
>     - Added unit tests in `DiffViewer.test.tsx` for undefined/null cases.
>     - Added integration tests in `integration.test.tsx` for malformed inputs.
>     - Added Storybook stories for visual testing of edge cases.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=humanlayer%2Fhumanlayer&utm_source=github&utm_medium=referral)<sup> for bd7adb56376a16cf4dbbdadd94176f2fe13d6076. You can [customize](https://app.ellipsis.dev/humanlayer/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: humanlayer-wui/scripts/test-malformed-inputs.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
```
Comment (dexhorthy) [inline humanlayer-wui/scripts/test-malformed-inputs.sh]: my instinct is that any bash script test should be a golang integration test, but for this, maybe even a storybook test where we hard-code the json

Comment (K-Mistele) [reply humanlayer-wui/scripts/test-malformed-inputs.sh]: yeah probably; just had a hard time reproducing it
